### PR TITLE
Add Intel Gaudi as a supported device.

### PR DESCRIPTION
**Pull request by vivekgoe**
This PR adds changes required to execute model training from torchtune repo on Intel Gaudi AI accelerators https://www.intel.com/content/www/us/en/products/details/processors/ai-accelerators/gaudi.html.

In order to use TorchTune with Intel Gaudi AI accelerators, Intel Gaudi Software Suite version v1.22.0 or higher is required. Instructions for configuring Intel Gaudi setup are available here https://docs.habana.ai/en/latest/Installation_Guide/index.html

Changes in this PR have been verified on Intel Gaudi2 and Intel Gaudi3 accelerators for LLama3.1-8B-LoRA and LLama3.1-70B-LoRA models. Reference commands for single epoch training for these models are shared below,
**LLama3.1-8B**
TORCH_COMPILE_BACKEND=hpu_backend tune run --nproc_per_node 8 lora_finetune_distributed --config recipes/configs/llama3_1/8B_lora.yaml tokenizer.max_seq_len=2048 dataset.packed=True compile=True device=hpu enable_activation_checkpointing=False batch_size=2 gradient_accumulation_steps=1 loss.mask_ignored_tokens=False optimizer.fused=False
**LLama3.1-70B**
TORCH_COMPILE_BACKEND=hpu_backend tune run --nproc_per_node 8 lora_finetune_distributed --config recipes/configs/llama3_1/70B_lora.yaml tokenizer.max_seq_len=2048 dataset.packed=True compile=True device=hpu loss.mask_ignored_tokens=False optimizer.fused=False

Reference command for eval using lm_eval on Intel Gaudi shared below,
export checkpoint_dir=/tmp/torchtune/llama3_1_8B/lora/epoch_0
tune run eleuther_eval --config eleuther_evaluation device=hpu checkpointer.model_type=LLAMA3 model._component_=torchtune.models.llama3_1.llama3_1_8b checkpointer.checkpoint_dir=$checkpoint_dir checkpointer.checkpoint_files=[$checkpoint_dir/model-00001-of-00004.safetensors,$checkpoint_dir/model-00002-of-00004.safetensors,$checkpoint_dir/model-00003-of-00004.safetensors,$checkpoint_dir/model-00004-of-00004.safetensors] tokenizer._component_=torchtune.models.llama3.llama3_tokenizer  tokenizer.path=$checkpoint_dir/original/tokenizer.model

### CODE REVIEW

**Comment by krammnic** [comment]
Hey! Thanks for the PR. Unfortunately, I don't have access to such accelerators. Would you mind to run few sanity checks in order to verify that everything is running correctly (the good way to share the results is a link to wandb with runs or some screenshots)?

**Comment by vivekgoe** [comment]
@krammnic Intel Gaudi accelerator SW does not have wandb integration as of now, therefore sharing training loss plots for 1 epoch running on 8xGaudi2 accelerators below,
image

Also adding eval harness results on fine-tuned LLama3.1-8B-LoRA checkpoint for "truthfulqa_mc2" task below,
Screenshot 2025-08-04 113610

Please let me know if additional data is required or if you have any follow-up questions. Thanks for reviewing this PR.
